bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:assert", "assert")
+#@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:ip", "ip")
+
+
+#@ flexibleServerName    = data.values.resource_name
+#@ dbSecretName          = data.values.resource_name
+#@ resourceGroupK8SName  = data.values.resource_name
+#@ serviceAccountName      = data.values.resource_name + "-reader"
+#@ serviceAccountNamespace = data.values.resource_namespace
+#@ roleName                = data.values.resource_name + "-reading"
+
+#@ mandatory_fields = [ "location", "resource_group", "databases", "firewall_rules" ]
```

Comment:
I think the problem is, that `ytt` does not support setting either `required` nor `enum` in the values schema. Thus we'd need to manage the package's openAPI schema manually as opposed to via `ytt -f .../config --data-values-schema-inspect -o openapi-v3`.

Reply:
Yeah this is always a bit annoying, openAPI schema has a much richer set of validation that we could use but isn't supported in generation. 

Personally I'd rather have custom written discoverable and strict jsonschema rather than runtime custom code doing this validation - but I don't have a strong opinion.

Reply:
> [...] openAPI schema has a much richer set of validation [...]

I agree.

> [...] that we could use but isn't supported in generation.

AFAIK it's also not supported for validation either. The openAPI schema is just a signal to and for consumption by "the outside world". When it comes to validation of package inputs, what counts is the values-schema & other runtime validation code. It might be, that some clients (the tanzu CLI?) do validation of inputs based on the openAPI schema, but when I manually create the `PackageInstall` with random input, the openAPI schema doesn't really mean a thing.

So if my above assumptions are correct, we could do something like:
- make the values-schema as strict as we need / want / can
- potentially use [`#@schema/validation` rules](https://carvel.dev/ytt/docs/develop/lang-ref-ytt-schema/#schemavalidation)
  - maybe additionally implement those rules in our code too, so they also apply on kapp-controllers that don't have `#@schema/validation` yet
- if we still / additionally want to make the openAPI schema more expressive / customize it, I'd look into still generating the schema with `ytt -f ... --data-values-schema-inspect -o openapi-v3`, but then apply an overlay, which changes some stuff, before we write the schema into the `Package`

I, however, also wanna state that I don't consider that blocking, regardless which way we go here; as long as the values-schema is reasonably "clean"/"correct"/"good" :)

Reply:
We check the validation rules, but even with setting the special env variable, the rules are only validated by the local KAPP CLI, not the Kapp Controller. So we figured it isn't worth it _at this time_.